share/availability/discovery/discovery.go

```diff
@@ -161,17 +181,49 @@ func (d *Discovery) ensurePeers(ctx context.Context) {
 				}
 				// listen to disconnect event to remove peer from set and reset backoff time
 				// reset timer in order to restart the discovery, once stored peer is disconnected
-				connStatus := e.(event.EvtPeerConnectednessChanged)
-				if connStatus.Connectedness == network.NotConnected {
-					if d.set.Contains(connStatus.Peer) {
-						log.Debugw("removing the peer from the peer set",
-							"peer", connStatus.Peer, "status", connStatus.Connectedness.String())
-						d.connector.RestartBackoff(connStatus.Peer)
-						d.set.Remove(connStatus.Peer)
-						d.onUpdatedPeers(connStatus.Peer, false)
-						d.host.ConnManager().UntagPeer(connStatus.Peer, topic)
-						t.Reset(d.discoveryInterval)
+				evnt := e.(event.EvtPeerConnectednessChanged)
+				switch evnt.Connectedness {
+				case network.NotConnected:
+					if !d.set.Contains(evnt.Peer) {
+						continue
+					}
+
+					d.host.ConnManager().UntagPeer(evnt.Peer, topic)
+					d.connector.RestartBackoff(evnt.Peer)
+					d.set.Remove(evnt.Peer)
+					d.onUpdatedPeers(evnt.Peer, false)
+					log.Debugw("removed peer from the peer set",
+						"peer", evnt.Peer, "status", evnt.Connectedness.String())
+				case network.Connected:
+					peerID := evnt.Peer
+					d.connectingLk.Lock()
+					cancelFind, ok := d.connecting[peerID]
+					d.connectingLk.Unlock()
+					if !ok {
+						continue
 					}
```

Comment:
Do we skip because it could be light client connected?

Reply:
Yes.